job_executor.go

```diff
@@ -245,7 +245,7 @@ func (e *jobExecutor) reportResult(ctx context.Context, res *jobExecutorResult)
 		return
 	}
 
-	if err := e.Completer.JobSetCompleted(e.JobRow.ID, e.stats, e.TimeNowUTC()); err != nil {
+	if err := e.Completer.JobSetState(e.stats, dbadapter.JobSetStateCompleted(e.JobRow.ID, e.TimeNowUTC())); err != nil {
```

Comment:
The invocations from the executor get a tiny bit more verbose, but not my that much.